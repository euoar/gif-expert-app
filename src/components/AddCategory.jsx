import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onValueChange = ({ target }) => {
    setInputValue(target.value)
  };

  const onSubmit = (event) =>{
    event.preventDefault();
    if (inputValue.trim().length >= 1) {
      onNewCategory(inputValue.trim());
      setInputValue('');
    }
  }
  return (
    <form onSubmit={onSubmit}>
      <input type="text" placeholder="Find gifs" value={inputValue} onChange={onValueChange}/>
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}