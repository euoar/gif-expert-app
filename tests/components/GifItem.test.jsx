import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItem component tests', () => {
  const title = 'Magnum PI';
  const url = 'https://media0.giphy.com/media/8Bl2AfBSAkzCc8dqQQ/giphy.gif?cid=1c21bd180zimt4ltf9ky2kifk7tpewr3el7ikxfwjhsm1bd5&ep=v1_gifs_search&rid=giphy.gif&ct=g';
  test('must mutch the snapshot', () => {
    const { container } = render(
      <GifItem
        title={title}
        url={url}
      />
    );

    expect(container).toMatchSnapshot();
  });
  
  test('must show the image with the provided url and alt', () => {
    render(
      <GifItem
        title={title}
        url={url}
      />
    );
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);

  });

  test('must show the component title', () => {
    render(
      <GifItem
        title={title}
        url={url}
      />
    );
    expect(screen.getByText(title)).toBeTruthy();

  });

})