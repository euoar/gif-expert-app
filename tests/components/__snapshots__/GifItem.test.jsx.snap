// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component tests must mutch the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Magnum PI"
      src="https://media0.giphy.com/media/8Bl2AfBSAkzCc8dqQQ/giphy.gif?cid=1c21bd180zimt4ltf9ky2kifk7tpewr3el7ikxfwjhsm1bd5&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Magnum PI
    </p>
  </div>
</div>
`;
